packages/hooks/src/useSet/index.ts

```diff
@@ -1,12 +1,9 @@
-import { useState } from 'react';
+import { useMemo, useState } from 'react';
 import useMemoizedFn from '../useMemoizedFn';
 
 function useSet<K>(initialValue?: Iterable<K>) {
-  const getInitValue = () => {
-    return initialValue === undefined ? new Set<K>() : new Set(initialValue);
-  };
-
-  const [set, setSet] = useState<Set<K>>(() => getInitValue());
+  const getInitValue = useMemo(() => new Set(initialValue), [initialValue]);
+  const [set, setSet] = useState<Set<K>>(getInitValue);
```

Comment:
u should not wrap it using `useMemo` because it's calculate eagerly when deps change, that is not expected behavior. Initial state should only generate once when component mounting, and should not re-calculate if input `initialValue` change. Just pass initailizer fn to `useState` like this.

```suggestion
  const [set, setSet] = useState<Set<K>>(() => new Set(initialValue));
```

Reply:
Yeh, It's no need to use `useMemo`, Just pass initializer fn to `useState`  @KangXinzhi

Reply:
Just this:

```ts
const getInitValue = () => new Set(initialValue);
const [set, setSet] = useState<Set<K>>(getInitValue);
```

Reply:
Thanks, I've made the required changes. 🙏